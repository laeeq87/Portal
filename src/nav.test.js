import React from "react";
import { render, screen, cleanup} from '@testing-library/react';
import Nav from "./nav";
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import { createMemoryHistory } from 'history'
import { Router } from "react-router";


afterEach(()=> {
    cleanup();
});

const props = {
    match:{
        params: jest.fn(),
    }
  }

it("Should Render Nav Component", ()=> {
    const history = createMemoryHistory(['/', '/login']);
    render(
        <Router history={history}>
            <Nav {...props} history={history} />
        </Router>
    );
    const homeComponentId  = screen.getByTestId('tst-nav')
    expect(homeComponentId).toBeInTheDocument();
}); 


test('matches snapshot', ()=> {
    const history = createMemoryHistory(['/', '/login']);
    const tree = renderer.create(
    <Router history={history}>
        <Nav {...props} history={history} />
    </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})
