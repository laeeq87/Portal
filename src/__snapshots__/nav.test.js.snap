// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="tst-nav"
>
  <nav>
    <div
      className="nav nav-tabs"
      id="nav-tab"
      role="tablist"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        aria-controls="nav-home"
        className="nav-link"
        role="tab"
      >
         
        <h5>
           My Account 
        </h5>
      </div>
      <a
        className="text-decoration-none"
        href="/account/profile"
        onClick={[Function]}
      >
        <button
          aria-controls="nav-profile"
          aria-selected="true"
          className="nav-link active"
          data-bs-target="#nav-user"
          data-bs-toggle="tab"
          id="nav-profile-tab"
          role="tab"
          type="button"
        >
          Profile
        </button>
      </a>
      <a
        className="text-decoration-none"
        href="/account/user"
        onClick={[Function]}
      >
        <button
          aria-controls="nav-user"
          aria-selected="true"
          className="nav-link "
          data-bs-target="#nav-profile"
          data-bs-toggle="tab"
          id="nav-user-tab"
          role="tab"
          type="button"
        >
          User
        </button>
      </a>
    </div>
  </nav>
  <div
    className="tab-content"
    id="nav-tabContent"
  >
    <div
      aria-labelledby="nav-profile-tab"
      className="tab-pane fade show active mt-5"
      id="nav-user"
      role="tabpanel"
    >
      <h3 />
    </div>
  </div>
</div>
`;
